celery-worker/replay_summary.py

```diff
@@ -194,6 +194,39 @@ def get_player_names(self) -> None:
                     except:
                         self.slot_to_player_name[slots] = "ERROR"
 
+    def get_game_duration(self) -> None:
+        """
+        Retrieves game duration, stored in duration.
+        :return None
+        """
+        self.duration = interval_data[0][-1]['time']
+
+    def get_GPM_and_XPM_and_CS(self) -> None:
+        """
+        Computes GPM and XPM and networth/xps/cs at each second.
+        Stores in slot_to_GPM, slot_to_XPM, slot_to_NWS, slot_to_XPS,
+        slot_to_LH, slot_to_deny, slot_to_LHS, slot_to_DENS
+        :return None:
+        """
+        self.slot_to_GPM = { 0:None, 1: None, 2: None, 3:None, 4: None, 5:None, 6:None, 7:None, 8:None, 9: None}
+        self.slot_to_XPM = { 0:None, 1: None, 2: None, 3:None, 4: None, 5:None, 6:None, 7:None, 8:None, 9: None}
+        self.slot_to_LH = { 0:None, 1: None, 2: None, 3:None, 4: None, 5:None, 6:None, 7:None, 8:None, 9: None}
+        self.slot_to_deny = { 0:None, 1: None, 2: None, 3:None, 4: None, 5:None, 6:None, 7:None, 8:None, 9: None}
+        self.slot_to_NWS = { 0:[], 1: [], 2: [], 3:[], 4:[], 5:[], 6:[], 7:[], 8:[], 9:[]}
+        self.slot_to_XPS = { 0:[], 1: [], 2: [], 3:[], 4:[], 5:[], 6:[], 7:[], 8:[], 9:[]}
+        self.slot_to_LHS = { 0:[], 1: [], 2: [], 3:[], 4:[], 5:[], 6:[], 7:[], 8:[], 9:[]}
+        self.slot_to_DENS= { 0:[], 1: [], 2: [], 3:[], 4:[], 5:[], 6:[], 7:[], 8:[], 9:[]}
+        for slot in self.slot_to_GPM:
+            self.slot_to_GPM[slot] = interval_data[slot][-1]['gold'] / (self.duration / 60)
+            self.slot_to_XPM[slot] = interval_data[slot][-1]['xp'] / (self.duration / 60)
+            self.slot_to_LH[slot] = interval_data[slot][-1]['lh']
+            self.slot_to_deny[slot] = interval_data[slot][-1]['denies']
+            for x in range(0, self.duration+1):
+                self.slot_to_NWS[slot].append(interval_data[slot][x]['networth'])
+                self.slot_to_XPS[slot].append(interval_data[slot][x]['xp'])
+                self.slot_to_LHS[slot].append(interval_data[slot][x]['lh'])
+                self.slot_to_deny[slot].append(inteval_data[slot][x]['denies'])
```

Comment:
```
self.slot_to_DENS[slot].append(interval_data[slot][x]['denies'])
```
I think this should be `slot_to_DENS` rather than `slot_to_deny` since we're trying to append to it, but I'm not 100% sure what the difference is between them

Reply:
I'll rename them here to make it clearer

Reply:
```suggestion
                self.slot_to_deny[slot].append(self.interval_data[slot][x]['denies'])
```
Thanks! There's also the same `self.interval_data` bit missing here as well.